### PR TITLE
Add abs tol support to comparison. Fix `sinc()`.

### PR DESCRIPTION
**Pull request by stuartarchibald**
`sinc()` was failing on a test machine via a bizarre numerical problem, it was
decided that supporting comparison by absolute tolerance may be useful
especially for values near zero, this patch introduces this feature. It also
fixes `sinc()` tests as a result. `sinc()` was failing for the value 1.e0 which
lead to `sin(pi)/pi` being evaluated, which naturally, due to numerical methods,
does not come out as 0.e0. The specific result value is below machine precision
and as demonstrated by failing code, somewhat machine/compiler/libm specific.

### CODE REVIEW

File: numba/tests/test_np_functions.py [hunk 1]
```diff
@@ -160,19 +164,34 @@ def test_sinc(self, flags=no_pyobj_flags):
         # See: https://github.com/numpy/numpy/pull/6699
         isoz = True
 
+        # Testing sinc(1.) leads to sin(pi)/pi, which is below machine
+        # precision in practice on most machines. Small floating point
+        # differences in sin() etc. may lead to large differences in the result
+        # that are at a range that in accessible using standard width
```
Comment (pitrou) [inline numba/tests/test_np_functions.py]: "in" is a typo, right?

Comment (stuartarchibald) [inline numba/tests/test_np_functions.py]: Indeed :)

File: numba/tests/support.py [hunk 2]
```diff
@@ -296,6 +307,19 @@ def _assertNumberEqual(first, second, delta=None):
...
             pass
 
+        # if absolute comparison is set, use it
+        if abs_tol is not None:
+            if abs_tol == "eps":
+                rtol = np.finfo(type(first)).eps
+            elif isinstance(abs_tol, float):
+                rtol = abs_tol
+            else:
+                self.fail(
```
Comment (pitrou) [inline numba/tests/support.py]: How about raising a ValueError instead? This is not a test failure, this is bug in a test.

Comment (stuartarchibald) [inline numba/tests/support.py]: I agree, will change.

**Comment by stuartarchibald** [comment]
Thanks for the review, patches in 0df9245

**Comment by pitrou** [comment]
Looking good, thank you!

**Pull request merged**